components/permissions_gates/team_permission_gate/team_permission_gate.jsx

```diff
@@ -7,7 +7,8 @@ import React from 'react';
 export default class TeamPermissionGate extends React.Component {
     shouldComponentUpdate(nextProps) {
         return this.props.invert !== nextProps.invert ||
-               this.props.hasPermission !== nextProps.hasPermission;
+               this.props.hasPermission !== nextProps.hasPermission ||
+               this.props.teamId !== nextProps.teamId;
```

Comment:
I assumed it's needed here too although this change wasn't fixing a specific bug that I observed.